proposals/z-index.md

```diff
@@ -0,0 +1,56 @@
+# z-index Convention
+
+
+## Problem
+We have no convention for choosing z-index. This means that developers can set an arbitrary z-index on a component that may have the side effect that it displays above or below another component of which it has no knowledge.
+
+
+## Solution
+
+### Base layers
+First we define the layers in an object.
+
+```js
+const layers = {
+  "-1": "-1",
+  "1": "1",
+  "2": "2",
+  "3": "3",
+  "4": "4",
+  "5": "5",
+  header: "101", // this includes, for example, the popover
+  modal: "102", // this includes the close button
```

Comment:
So is the idea that these 8 layers are the only z-index layers that will be permitted in an app?

Reply:
Yep. If they turn out to not be enough, the object should be modified. But for now, trying to limit the use of arbitrary z-index values.

Reply:
Okay, if that's the case it would also be good to lint app styles to ensure that `z-index` is not directly used, although we are not currently linting them at all I don't think

Reply:
@dbbk using  https://github.com/styled-components/stylelint-processor-styled-components?